File: parallel_test.js
const pathik = require('pathik');
const path = require('path');
const fs = require('fs');

async function main() {
  // Create output directories
  const baseDir = path.resolve('./output_test_js');
  const parallelDir = path.join(baseDir, 'parallel');
  const sequentialDir = path.join(baseDir, 'sequential');
  
  fs.mkdirSync(parallelDir, { recursive: true });
  fs.mkdirSync(sequentialDir, { recursive: true });
  
  // Test URLs - mix of different sites
  const urls = [
    'https://example.com',
    'https://httpbin.org/html',
    'https://jsonplaceholder.typicode.com',
    'https://books.toscrape.com',
    'https://quotes.toscrape.com'
  ];
  
  console.log(`Testing with ${urls.length} URLs...`);
  
  // Test parallel crawling
  console.log('\n=== PARALLEL CRAWLING ===');
  const parallelStart = Date.now();
  
  try {
    const parallelResults = await pathik.crawl(urls, { 
      outputDir: parallelDir,
      parallel: true
    });
    
    const parallelTime = (Date.now() - parallelStart) / 1000;
    console.log(`Parallel crawling completed in ${parallelTime.toFixed(2)} seconds`);
    
    // Print results summary
    for (const [url, info] of Object.entries(parallelResults)) {
      const status = info.html && info.markdown ? '✅ Success' : '❌ Failed';
      console.log(`  ${url}: ${status}`);
    }
    
    // Test sequential crawling
    console.log('\n=== SEQUENTIAL CRAWLING ===');
    const sequentialStart = Date.now();
    
    try {
      const sequentialResults = await pathik.crawl(urls, { 
        outputDir: sequentialDir,
        parallel: false
      });
      
      const sequentialTime = (Date.now() - sequentialStart) / 1000;
      console.log(`Sequential crawling completed in ${sequentialTime.toFixed(2)} seconds`);
      
      // Print results summary
      for (const [url, info] of Object.entries(sequentialResults)) {
        const status = info.html && info.markdown ? '✅ Success' : '❌ Failed';
        console.log(`  ${url}: ${status}`);
      }
      
      // Compare performance
      console.log('\n=== PERFORMANCE COMPARISON ===');
      if (parallelTime < sequentialTime) {
        const speedup = sequentialTime / parallelTime;
        console.log(`Parallel crawling was ${speedup.toFixed(2)}x faster than sequential crawling`);
      } else {
        console.log(`Warning: Parallel crawling was not faster in this test`);
      }
      
      console.log(`Parallel: ${parallelTime.toFixed(2)}s vs Sequential: ${sequentialTime.toFixed(2)}s`);
      console.log(`\nOutput files are located in: ${baseDir}`);
    } catch (error) {
      console.error(`Error during sequential crawling: ${error.message}`);
      process.exit(1);
    }
  } catch (error) {
    console.error(`Error during parallel crawling: ${error.message}`);
    process.exit(1);
  }
}

main().catch(console.error); 

File: parallel_test.py
import pathik
import os
import time
import sys

def main():
    # Create output directories
    base_dir = os.path.abspath("./output_test")
    parallel_dir = os.path.join(base_dir, "parallel")
    sequential_dir = os.path.join(base_dir, "sequential")
    
    os.makedirs(parallel_dir, exist_ok=True)
    os.makedirs(sequential_dir, exist_ok=True)
    
    # Test URLs - mix of different sites
    urls = [
        "https://example.com",
        "https://httpbin.org/html",
        "https://jsonplaceholder.typicode.com",
        "https://books.toscrape.com",
        "https://quotes.toscrape.com"
    ]
    
    print(f"Testing with {len(urls)} URLs...")
    
    # Test parallel crawling
    print("\n=== PARALLEL CRAWLING ===")
    parallel_start = time.time()
    
    try:
        parallel_results = pathik.crawl(urls, output_dir=parallel_dir, parallel=True)
        parallel_time = time.time() - parallel_start
        
        print(f"Parallel crawling completed in {parallel_time:.2f} seconds")
        
        # Print results summary
        for url, info in parallel_results.items():
            status = "✅ Success" if info.get("html") and info.get("markdown") else "❌ Failed"
            print(f"  {url}: {status}")
    except Exception as e:
        print(f"Error during parallel crawling: {e}")
        sys.exit(1)
    
    # Test sequential crawling
    print("\n=== SEQUENTIAL CRAWLING ===")
    sequential_start = time.time()
    
    try:
        sequential_results = pathik.crawl(urls, output_dir=sequential_dir, parallel=False)
        sequential_time = time.time() - sequential_start
        
        print(f"Sequential crawling completed in {sequential_time:.2f} seconds")
        
        # Print results summary
        for url, info in sequential_results.items():
            status = "✅ Success" if info.get("html") and info.get("markdown") else "❌ Failed"
            print(f"  {url}: {status}")
    except Exception as e:
        print(f"Error during sequential crawling: {e}")
        sys.exit(1)
    
    # Compare performance
    if parallel_time < sequential_time:
        speedup = sequential_time / parallel_time
        print(f"\n=== PERFORMANCE COMPARISON ===")
        print(f"Parallel crawling was {speedup:.2f}x faster than sequential crawling")
        print(f"Parallel: {parallel_time:.2f}s vs Sequential: {sequential_time:.2f}s")
    else:
        print("\n=== PERFORMANCE COMPARISON ===")
        print(f"Warning: Parallel crawling was not faster in this test")
        print(f"Parallel: {parallel_time:.2f}s vs Sequential: {sequential_time:.2f}s")
    
    print(f"\nOutput files are located in: {base_dir}")

if __name__ == "__main__":
    main() 